### PR TITLE
Failing test for class name resolution in class constants

### PR DESCRIPTION
**Pull request by AydinHassan**
Hi, 

Class name resolution in constants doesn't seem to work, I've provided a failing test case but I have no idea how to fix it :smile: 

If you could give some pointers I can have a pop at fixing it?

### CODE REVIEW

File: test/unit/NodeCompiler/CompileNodeToValueTest.php [hunk 1]
```diff
@@ -251,4 +251,37 @@ public function method($param = static::BAR) {}
...
+
+        class Foo {
+        }
+        class Bat {
+            const QUX = Foo::class;
+        }
+        ';
+
+        $reflector = new ClassReflector(new StringSourceLocator($phpCode));
+        $classInfo = $reflector->reflect('Bat');
```
Comment (Ocramius) [inline test/unit/NodeCompiler/CompileNodeToValueTest.php]: Shouldn't this be `Bar\Bat`?

Comment (Ocramius) [inline test/unit/NodeCompiler/CompileNodeToValueTest.php]: @asgrim should this method throw an exception on not found class?

Comment (AydinHassan) [inline test/unit/NodeCompiler/CompileNodeToValueTest.php]: @Ocramius the other test should have been `Bar\Bat` this one isn't namespaced. Fixed that now.

**Comment by asgrim** [comment]
Thanks @AydinHassan - I've opened a new PR with your test and the implementation in #168 - much appreciated

**Comment by AydinHassan** [comment]
Awesome, thanks @asgrim !

**Pull request closed without merging**